saviSimulation/savi/jason_processing/SAVIWorld_model.java

```diff
@@ -105,82 +108,93 @@ public void setup() {
 		pause = loadShape("SimImages/pause.svg");
-		restart = loadShape("SimImages/replay.svg");
 
 		// load images for visualization
+		restart = loadShape("SimImages/replay.svg");
 		treeImage = loadShape("SimImages/tree.svg");
 		houseImage = loadShape("SimImages/home.svg");
 		play = loadShape("SimImages/play.svg");
 		pause = loadShape("SimImages/pause.svg");
 		restart = loadShape("SimImages/replay.svg");
 		threatImage = loadShape("SimImages/warning.svg");
-		uasImage = loadShape("SimImages/robot.svg");
-		antennaImage = loadShape("SimImages/antenna.svg"); //TODO: get an antenna image
+		ugvImage = loadShape("SimImages/robot.svg");
+		uavImage = loadShape("SimImages/airplane.svg");
+		antennaImage = loadShape("SimImages/antenna.svg");
 
 		Random rand = new Random();
-
-		for (int i = 0; i < NUMBER_UAS; i++) { // Put UAS
-			// _PIXELS is the maximum and the 1 is our minimum
-			// TODO: right now agents are initialized with strings "0", "1", "2", ... as
-			// identifiers and a fixed type "demo" which matches their asl file name. This
-			// should be configurable...
-			if (RANDOM_SEED != -1) {
-				rand = new Random(RANDOM_SEED + i);
+		
+				
+		for(int i = 0; i < NUMBER_UAV+NUMBER_UGV; i++)  { //Put UxV
+			//_PIXELS is the maximum and the 1 is our minimum
+			//TODO: right now agents are initialized with strings "0", "1", "2", ... as identifiers and a fixed type "demo" which matches their asl file name. This should be configurable...
+			if(RANDOM_SEED != -1) {
+				rand = new Random(RANDOM_SEED+i);
+			}	
+		
+			if(i < NUMBER_UGV)  { //Put UgV
+				//_PIXELS is the maximum and the 1 is our minimum
+				//TODO: right now agents are initialized with strings "0", "1", "2", ... as identifiers and a fixed type "demo" which matches their asl file name. This should be configurable...
+				UgV ugv= new UgV(i, new PVector(rand.nextInt(X_PIXELS) + 1, rand.nextInt(Y_PIXELS) + 1, UGV_SIZE/2), UGV_SIZE/2,"demo", this, ugvImage, REASONING_CYCLE_PERIOD, "robot");
+				wifiParticipants.add(ugv.getAntennaRef());
+				objects.add(ugv);
+			}else {
+				UaV uav = new UaV(i, new PVector(rand.nextInt(X_PIXELS) + 1, rand.nextInt(Y_PIXELS) + 1, UAV_SIZE/2), UAV_SIZE/2,"demo", this, uavImage, REASONING_CYCLE_PERIOD, "airplane");
+				wifiParticipants.add(uav.getAntennaRef());
+				objects.add(uav);
 			}
-			UAS uas = new UAS(i, new PVector(rand.nextInt(X_PIXELS) + 1, rand.nextInt(Y_PIXELS) + 1), UAS_SIZE,
-					"demo", this, uasImage, REASONING_CYCLE_PERIOD);
-			objects.add(uas);
-			wifiParticipants.add(uas.getAntennaRef());
 			
 		}
+		
+		
 
 		for (int i = 0; i < NUMBER_TREES; i++) { // Put trees
 			// _PIXELS is the maximum and the 1 is our minimum.
 			if (RANDOM_SEED != -1) {
 				rand = new Random(2 * RANDOM_SEED + i);
 			}
-			objects.add(new WorldObject(i, new PVector(rand.nextInt(X_PIXELS) + 1, rand.nextInt(Y_PIXELS) + 1),
+			objects.add(new WorldObject(i, new PVector(rand.nextInt(X_PIXELS) + 1, rand.nextInt(Y_PIXELS) + 1, TREE_SIZE/2),
```

Comment:
every object seems to have a radius equal to half its size. Does it make sense to have a separate parameter for the radius?

Reply:
We used 1/2 of the size to make sure that the size in the image is the size of the object for the simulation. Otherwise, when we visualize it may seem that the UxV is not doing object avoidance properly

Reply:
Having the image size we can calculate the radius (also notice that for objects in the ground the position at its z coordinate will be the radius).

Reply:
this isn't the radius in the constructor, it's the z coordinate.